src/main/java/ua/vedroid/dao3/dao/jdbc/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,206 @@
+package ua.vedroid.dao3.dao.jdbc;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import ua.vedroid.dao3.dao.CarDao;
+import ua.vedroid.dao3.exceptions.DataProcessingException;
+import ua.vedroid.dao3.lib.Dao;
+import ua.vedroid.dao3.model.Car;
+import ua.vedroid.dao3.model.Driver;
+import ua.vedroid.dao3.model.Manufacturer;
+import ua.vedroid.dao3.util.ConnectionUtil;
+
+@Dao
+public class CarDaoJdbcImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String query = "INSERT INTO cars(m_id, model) VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement =
+                        connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setLong(1, car.getManufacturer().getId());
+            statement.setString(2, car.getModel());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getLong("id"));
+            }
+            if (!car.getDrivers().isEmpty()) {
+                addDrivers(connection, car);
+            }
+            return car;
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Insert " + car + " to database is failed", ex);
+        }
+    }
+
+    @Override
+    public Optional<Car> getById(Long id) {
+        String query = "SELECT * FROM cars "
+                + "JOIN manufacturers m on m.id = cars.m_id "
+                + "LEFT JOIN cars_drivers cd on cars.id = cd.car_id "
+                + "LEFT JOIN drivers d on d.id = cd.driver_id "
+                + "WHERE cars.id = ?"
+                + " AND cars.is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(getCarFromResultSet(connection, resultSet));
+            }
+            return Optional.empty();
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Get car with id="
+                    + id + " is failed", ex);
+        }
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT * FROM cars JOIN manufacturers m on m.id = cars.m_id "
```

Comment:
I was told that star select is a bad practice. TBH not sure about that but be advised.

Reply:
@Vedroid @Artificial-Friend The thing is that you may add new fields to your table later, so it's better to specify here which exactly field you want to select(even though now its all of them);

Reply:
Now I get it, thank you!